maingate/src/instructions.rs

```diff
@@ -30,69 +28,22 @@ use crate::{AssignedCondition, AssignedValue};
 /// `... + assigned_or_unassigned_witness_0 * fixed_0 +
 /// assigned_or_unassigned_witness_1 * fixed_1 + ... `
 #[derive(Clone)]
-pub enum Term<F: FieldExt> {
+pub enum Term<'a, F: FieldExt> {
```

Comment:
This might complicate a lot the syntax and force to always declare the lifetime when indeed `AssignedValue` doesn't really take that much space.

It's fine if we want to merge like this. But I'd not abuse this and use it only when it really gives significant savings.

Reply:
After checking with @ed255 It's fine to leave this as is as removes a lot of `clone`s and also does not introduce a lot of lifetime management hassle.